staging/src/k8s.io/apimachinery/pkg/util/json/json.go

```diff
@@ -119,3 +120,20 @@ func convertNumber(n json.Number) (interface{}, error) {
 	// An overflow will return an error
 	return n.Float64()
 }
+
+// Sniff does a constant-time inspection of the provided bytes and returns whether or not they may
+// contain JSON. If unknown is true, the determination could be wrong.
+func Sniff(src []byte) (json bool, unknown bool) {
```

Comment:
the double-bool signature is awkward... json really means `possiblyJSON`, right? if `unknown` is true, does it ever make sense to look at the `json` bool? if not, this could become `PossiblyJSON(src []byte) bool`, where we only return false when we see a definitively non-json character?